[GH-257] Android Application ID と iOS Bundle ID を別々にリネームできるようにする

daichikuwa0618 opened the pull request:
## 概要

resolve #257 

設定をまとめたファイルを作ってパースさせる案もあったが、一回限りのスクリプトに割くには見合っていないということで最小限の変更で対応。
#203 の対応時にあらためて検討されたい。

## レビュー観点

- 実行後の変更箇所・変更内容に過不足がないか

## レビューレベル

- [ ] Lv1: ぱっとみて違和感がないかチェックして Approve する
- [ ] Lv2: 仕様レベルまで理解して、仕様通りに動くかある程度検証して Approve する
- [x] Lv3: 実際に環境で動作確認したうえで Approve する

## レビュー優先度

- [ ] すぐに見てもらいたい ( hotfix など ) 🚀
- [ ] 今日中に見てもらいたい 🚗
- [ ] 今日〜明日中で見てもらいたい 🚶
- [x] 数日以内で見てもらいたい 🐢

## 確認したこと

- [x] 引数が 0, 1 のときはエラーが出ること
- [x] 修正前後のスクリプト実行後差分に差異がないこと (以下参照)

<details><summary><b>参考: 修正前の実行後</b></summary><div>

```diff
diff --git a/apps/app/android/app/src/main/kotlin/jp/co/yumemi/flutter_app/MainActivity.kt b/apps/app/android/app/src/main/kotlin/jp/co/yumemi/flutter_app/MainActivity.kt
deleted file mode 100644
index ccc7325..0000000
--- a/apps/app/android/app/src/main/kotlin/jp/co/yumemi/flutter_app/MainActivity.kt
+++ /dev/null
@@ -1,5 +0,0 @@
-package jp.co.yumemi.flutter_app
-
-import io.flutter.embedding.android.FlutterActivity
-
-class MainActivity: FlutterActivity()
diff --git a/apps/app/flavor/dev.json b/apps/app/flavor/dev.json
index 5ba9ec3..bf98f95 100644
--- a/apps/app/flavor/dev.json
+++ b/apps/app/flavor/dev.json
@@ -1,6 +1,6 @@
 {
     "flavor": "dev",
     "appName": "dev-Template",
-    "appId": "jp.co.yumemi.template",
+    "appId": "com.example.rename",
     "appIdSuffix": ".dev"
 }
diff --git a/apps/app/flavor/prd.json b/apps/app/flavor/prd.json
index c6dcb85..2929e00 100644
--- a/apps/app/flavor/prd.json
+++ b/apps/app/flavor/prd.json
@@ -1,6 +1,6 @@
 {
     "flavor": "prd",
     "appName": "Template",
-    "appId": "jp.co.yumemi.template",
+    "appId": "com.example.rename",
     "appIdSuffix": ""
 }
diff --git a/apps/app/flavor/stg.json b/apps/app/flavor/stg.json
index 0e7698c..d3722da 100644
--- a/apps/app/flavor/stg.json
+++ b/apps/app/flavor/stg.json
@@ -1,6 +1,6 @@
 {
     "flavor": "stg",
     "appName": "stg-Template",
-    "appId": "jp.co.yumemi.template",
+    "appId": "com.example.rename",
     "appIdSuffix": ".stg"
 }
```

</div></details>

修正後の実行後差分

```sh
cd tools
sh rename-application-id.sh com.example.android.rename com.example.ios.rename
```

```diff
diff --git a/apps/app/android/app/src/main/kotlin/jp/co/yumemi/flutter_app/MainActivity.kt b/apps/app/android/app/src/main/kotlin/jp/co/yumemi/flutter_app/MainActivity.kt
deleted file mode 100644
index ccc7325..0000000
--- a/apps/app/android/app/src/main/kotlin/jp/co/yumemi/flutter_app/MainActivity.kt
+++ /dev/null
@@ -1,5 +0,0 @@
-package jp.co.yumemi.flutter_app
-
-import io.flutter.embedding.android.FlutterActivity
-
-class MainActivity: FlutterActivity()
diff --git a/apps/app/flavor/dev.json b/apps/app/flavor/dev.json
index a1c7dae..42c02b6 100644
--- a/apps/app/flavor/dev.json
+++ b/apps/app/flavor/dev.json
@@ -1,7 +1,7 @@
 {
     "flavor": "dev",
     "appName": "dev-Template",
-    "appIdAndroid": "jp.co.yumemi.template",
-    "appIdIos": "jp.co.yumemi.template",
+    "appIdAndroid": "com.example.android.rename",
+    "appIdIos": "com.example.ios.rename",
     "appIdSuffix": ".dev"
 }
diff --git a/apps/app/flavor/prd.json b/apps/app/flavor/prd.json
index 189021c..3d18465 100644
--- a/apps/app/flavor/prd.json
+++ b/apps/app/flavor/prd.json
@@ -1,7 +1,7 @@
 {
     "flavor": "prd",
     "appName": "Template",
-    "appIdAndroid": "jp.co.yumemi.template",
-    "appIdIos": "jp.co.yumemi.template",
+    "appIdAndroid": "com.example.android.rename",
+    "appIdIos": "com.example.ios.rename",
     "appIdSuffix": ""
 }
diff --git a/apps/app/flavor/stg.json b/apps/app/flavor/stg.json
index 00f0eea..d4a2a7c 100644
--- a/apps/app/flavor/stg.json
+++ b/apps/app/flavor/stg.json
@@ -1,7 +1,7 @@
 {
     "flavor": "stg",
     "appName": "stg-Template",
-    "appIdAndroid": "jp.co.yumemi.template",
-    "appIdIos": "jp.co.yumemi.template",
+    "appIdAndroid": "com.example.android.rename",
+    "appIdIos": "com.example.ios.rename",
     "appIdSuffix": ".stg"
 }
```

## 動作確認手順

```sh
cd tools
sh rename-application-id.sh com.example.android.rename com.example.ios.rename
```

## 備考

<!--
参考文献などがあれば記載してください。
また、マージするタイミングが特殊という注意事項などあればあわせて記載してください。
-->

daichikuwa0618 commented on .github/initialization/RENAME_APPLICATION_ID.md:
差分も確認いただきたいですが、 https://github.com/yumemi-inc/flutter-mobile-project-template/blob/feature/GH-257/.github/initialization/RENAME_APPLICATION_ID.md で完成後のドキュメントが問題ないかも見ていただきたいです。

daichikuwa0618 commented on apps/app/flavor/dev.json:
```diff
@@ -1,6 +1,7 @@
     "flavor": "dev",
     "appName": "dev-Template",
-    "appId": "jp.co.yumemi.template",
+    "appIdAndroid": "jp.co.yumemi.template",
+    "appIdIos": "jp.co.yumemi.template",
```
`Ios` はかなり気持ち悪いが、、、、

- `androidAppId` / `iosAppId` はアルファベット順で並べたときに離れてしまう
- lower camel case 遵守

ということでこの命名を採用してます。

daichikuwa0618 commented on tools/rename-application-id.sh:
```diff
@@ -8,16 +8,18 @@ set -e
 
 echo "Starting application ID renaming process..."
 
-readonly NEW_APPLICATION_ID=$1
-readonly OLD_APPLICATION_ID="jp.co.yumemi.template"
```
削除した 2 変数が使われてないことは grep で確認済み

daichikuwa0618 commented on .github/initialization/RENAME_APPLICATION_ID.md:
````diff
@@ -14,13 +14,17 @@
 1. スクリプトの実行に必要な権限があることを確認してください。
-2. スクリプトの実行時に、新しいアプリケーションIDを引数として指定する。
-
-```bash
-bash rename-application-id.sh "com.example.rename"
````
macOS のデフォルトは zsh だし、シェルスクリプトのシバンも `/bin/sh` なので bash に限定しなくてもいいかなと。

K9i-0 reviewed (changes requested):
修正に関してはLGTM
既存がおかしいのでSlackにコメントしました
https://yumemi.slack.com/archives/C0585H2BG9L/p1718275216573499

daichikuwa0618 commented:
@K9i-0 さんに Slack にて指摘してもらった点:

1. 既存の Doc の日本語がおかしい部分がある
2. `jp` 始まりの ID に replace しようとすると新しい `MainActivity.kt` も削除されてしまう
   - path の全てに一致させないで、 `jp` 配下すべて削除していたため

でした。

1 については bf4d290733ce0cfd962713d375ed95ba50ecb3be に混ぜて修正しました (分けると良くわからないまとまりになりそうだったので)。
2 については 0f63f12c8767e3ce99a827c8fa7dd25a14f47c7d にて修正しました。Step 3 と 4 を入れ替えて、先に削除 → 追加という手順に変更しています。
動作確認として、 `jp` 始まりの ID に replace したときの before / after の diff です。

<details><summary><b>diff before</b></summary><div>

```diff
diff --git a/apps/app/android/app/src/main/kotlin/jp/co/yumemi/flutter_app/MainActivity.kt b/apps/app/android/app/src/main/kotlin/jp/co/yumemi/flutter_app/MainActivity.kt
deleted file mode 100644
index ccc7325..0000000
--- a/apps/app/android/app/src/main/kotlin/jp/co/yumemi/flutter_app/MainActivity.kt
+++ /dev/null
@@ -1,5 +0,0 @@
-package jp.co.yumemi.flutter_app
-
-import io.flutter.embedding.android.FlutterActivity
-
-class MainActivity: FlutterActivity()
diff --git a/apps/app/flavor/dev.json b/apps/app/flavor/dev.json
index 5ba9ec3..6b06e89 100644
--- a/apps/app/flavor/dev.json
+++ b/apps/app/flavor/dev.json
@@ -1,6 +1,6 @@
 {
     "flavor": "dev",
     "appName": "dev-Template",
-    "appId": "jp.co.yumemi.template",
+    "appId": "jp.example.replaced",
     "appIdSuffix": ".dev"
 }
diff --git a/apps/app/flavor/prd.json b/apps/app/flavor/prd.json
index c6dcb85..6b0ff02 100644
--- a/apps/app/flavor/prd.json
+++ b/apps/app/flavor/prd.json
@@ -1,6 +1,6 @@
 {
     "flavor": "prd",
     "appName": "Template",
-    "appId": "jp.co.yumemi.template",
+    "appId": "jp.example.replaced",
     "appIdSuffix": ""
 }
diff --git a/apps/app/flavor/stg.json b/apps/app/flavor/stg.json
index 0e7698c..857f27b 100644
--- a/apps/app/flavor/stg.json
+++ b/apps/app/flavor/stg.json
@@ -1,6 +1,6 @@
 {
     "flavor": "stg",
     "appName": "stg-Template",
-    "appId": "jp.co.yumemi.template",
+    "appId": "jp.example.replaced",
     "appIdSuffix": ".stg"
 }
```

</div></details>

<details><summary><b>diff after</b></summary><div>

```diff
diff --git a/apps/app/android/app/src/main/kotlin/jp/co/yumemi/flutter_app/MainActivity.kt b/apps/app/android/app/src/main/kotlin/jp/co/yumemi/flutter_app/MainActivity.kt
deleted file mode 100644
index ccc7325..0000000
--- a/apps/app/android/app/src/main/kotlin/jp/co/yumemi/flutter_app/MainActivity.kt
+++ /dev/null
@@ -1,5 +0,0 @@
-package jp.co.yumemi.flutter_app
-
-import io.flutter.embedding.android.FlutterActivity
-
-class MainActivity: FlutterActivity()
diff --git a/apps/app/android/app/src/main/kotlin/jp/example/and/MainActivity.kt b/apps/app/android/app/src/main/kotlin/jp/example/and/MainActivity.kt
index e69de29..a719307 100644
--- a/apps/app/android/app/src/main/kotlin/jp/example/and/MainActivity.kt
+++ b/apps/app/android/app/src/main/kotlin/jp/example/and/MainActivity.kt
@@ -0,0 +1,5 @@
+package jp.example.and
+
+import io.flutter.embedding.android.FlutterActivity
+
+class MainActivity: FlutterActivity()
diff --git a/apps/app/flavor/dev.json b/apps/app/flavor/dev.json
index a1c7dae..c87ba18 100644
--- a/apps/app/flavor/dev.json
+++ b/apps/app/flavor/dev.json
@@ -1,7 +1,7 @@
 {
     "flavor": "dev",
     "appName": "dev-Template",
-    "appIdAndroid": "jp.co.yumemi.template",
-    "appIdIos": "jp.co.yumemi.template",
+    "appIdAndroid": "jp.example.and",
+    "appIdIos": "jp.example.ios",
     "appIdSuffix": ".dev"
 }
diff --git a/apps/app/flavor/prd.json b/apps/app/flavor/prd.json
index 189021c..0809680 100644
--- a/apps/app/flavor/prd.json
+++ b/apps/app/flavor/prd.json
@@ -1,7 +1,7 @@
 {
     "flavor": "prd",
     "appName": "Template",
-    "appIdAndroid": "jp.co.yumemi.template",
-    "appIdIos": "jp.co.yumemi.template",
+    "appIdAndroid": "jp.example.and",
+    "appIdIos": "jp.example.ios",
     "appIdSuffix": ""
 }
diff --git a/apps/app/flavor/stg.json b/apps/app/flavor/stg.json
index 00f0eea..b0743a9 100644
--- a/apps/app/flavor/stg.json
+++ b/apps/app/flavor/stg.json
@@ -1,7 +1,7 @@
 {
     "flavor": "stg",
     "appName": "stg-Template",
-    "appIdAndroid": "jp.co.yumemi.template",
-    "appIdIos": "jp.co.yumemi.template",
+    "appIdAndroid": "jp.example.and",
+    "appIdIos": "jp.example.ios",
     "appIdSuffix": ".stg"
 }
```

</div></details>

K9i-0 reviewed (approved):
LGTM

PRと関係ない修正までありがとうございます 🙇
